elastic/kibana
# [Uptime] Allow users to save uptime settings with _inspect on

## Summary

This PR fixes #116368.

It allows users to save uptime settings when they've got the advanced setting to inspect ES queries turned on.

Previously, it was not possible to save these settings because the validation within that route would not allow for a _inspect field to be passed.

Now, the `_inspect` property will be validated as optional but ignored when saving the configuration as a saved object.

### How to test this PR

In addition to the functional test added to validate that this route handles the `_inspect` property correctly, this PR should be E2E tested following the steps below:

1. On your Kibana instance, access Stack Management > Advanced settings and turn on the option to `inspect` ES queries, as shown in the screenshot below.
    ![Screenshot 2021-11-19 at 10 57 15](https://user-images.githubusercontent.com/6868147/142612974-73a1ee6d-0ee9-4dbc-91be-b73d58de3aaf.png)
2. Go to the `Uptime` app, and, on the top right, open the settings view.
3. Make any change to the settings, like adding or removing a default connector, for example.
4. Save your changes and note there's a toast saying the request succeeded, whereas before it would fail.
    ![Screenshot 2021-11-19 at 11 03 34](https://user-images.githubusercontent.com/6868147/142612838-cefea9b9-5ac5-4ebd-8a1d-afc6a07df438.png)
5. Go to Stack Management > Saved Objects, find the `Uptime Settings` saved object, and inspect it. There, you should confirm that the `_inspect` flag is _not_ saved.
    ![Screenshot 2021-11-19 at 11 07 43](https://user-images.githubusercontent.com/6868147/142612893-4779f715-8d4c-41d3-8e49-014bf4e335e0.png)




### Checklist

Delete any items that are not applicable to this PR.


### For maintainers


## Release note

Fixes a bug which would prevent users from saving Uptime configurations when the `inspect` option was turned on.
